RuddockWebsite/__init__.py

```diff
@@ -29,18 +30,36 @@
 @app.before_request
 def before_request():
   ''' Logic executed before request is processed. '''
-
   # Connect to the database and publish it in flask.g
   g.db = engine.connect()
 
 @app.teardown_request
 def teardown_request(exception):
   ''' Logic executed after every request is finished. '''
-
   # Close database connection.
   db = getattr(g, 'db', None)
   if db is not None:
     db.close()
 
+# Error handlers
+@app.errorhandler(404)
+def page_not_found(error):
+  ''' Handles a 404 page not found error. '''
+  return render_template("404.html"), 404
+
+@app.errorhandler(500)
+def internal_server_error(error):
+  '''
+  Handles a 500 internal server error response. This error is usually the
+  result of an improperly configured server or bugs in the actual codebase
+  (user errors should be handled gracefully), so IMSS must be notified if this
+  error occurs.
+  '''
+  msg = email_templates.ErrorCaughtEmail.format(traceback.format_exc())
+  subject = "Ruddock website error"
+  to = "imss@ruddock.caltech.edu"
+  email_utils.sendEmail(to, msg, subject)
```

Comment:
Can you not send the email if debug is on? This will prevent similar spam to what we got before, and would make sense if a developer is currently working on something that inevitably leads to 500 errors.